tensor_genn/algorithms/relu_ann.py

```diff
@@ -10,18 +10,67 @@ def __init__(self,neuron_resting_voltage=-60.0,neuron_threshold_voltage=-55.0,
         membrane_capacitance=1.0, model_timestep=1.0, single_example_time=350.):
         self.Vres = neuron_resting_voltage
         self.Vthr = neuron_threshold_voltage
-        self.Cm = membrane_capacitance
+        self.Cm = membrane_capacitance # For DNN - 1.0, CNN - 0.4
         self.timestep = model_timestep
         self.single_example_time = single_example_time
 
+    def create_weight_matrices(self, tf_model):
+        tf_weights = tf_model.get_weights()
+        tf_layers = tf_model.layers
+
+        g_model_weights = []
+        n_units = [np.prod(tf_layers[0].input_shape[1:])]
+        i = j = 1
+
+        for layer in tf_layers:
+            if not isinstance(layer, tf.keras.layers.Flatten):
+                n_units.append(np.prod(layer.output_shape[1:]))
+                syn_weights = np.zeros((n_units[j-1],n_units[j]))
+
+                if isinstance(layer,tf.keras.layers.Dense):
+                    syn_weights = tf_weights[i-1]
+                    i += 1
+
+                elif isinstance(layer,tf.keras.layers.Conv2D):
+                    kw,kh = layer.kernel_size # 2,2
+                    sw,sh = layer.strides # 1,1
+                    ih,iw,ic = layer.input_shape[1:] # 3,3,2
+                    oh,ow,oc = layer.output_shape[1:] # 2,2,2
+        
+                    for n in range(int(n_units[j])): # output unit/conv number
+                        for k in range(kh): # kernel height
+                            syn_weights[((n//oc)%ow)*ic*sw + ((n//oc)//ow)*ic*iw*sh + k*ic*iw:
+                                        ((n//oc)%ow)*ic*sw + ((n//oc)//ow)*ic*iw*sh + k*ic*iw + kw*ic,
+                                        n] = tf_weights[i-1][k,:,:,n%oc].reshape((-1))
+
+                    i += 1
+
+                elif isinstance(layer,tf.keras.layers.AveragePooling2D):
```

Comment:
Similarly, some more commenting on how this works would be really useful. Also, doesn't this approach come from a paper? If so it's always good to add a reference to the comments (also anywhere else this is the case)

Reply:
This approach has been used in many papers and isn't taken from one in particular. Could I just add a reference to the ReLU ANN paper at the top of the file?

Reply:
sounds good